Stack/Opc.Ua.Core/Security/Certificates/CertificateIdentifier.cs

```diff
@@ -487,8 +503,189 @@ public ICertificateStore OpenStore()
             store.Open(this.StorePath, false);
             return store;
         }
+
+        /// <summary>
+        /// Retrieves the minimum accepted key size given the security configuration
+        /// </summary>
+        /// <param name="securityConfiguration"></param>
+        /// <returns></returns>
+        public ushort GetMinKeySize(SecurityConfiguration securityConfiguration)
+        {
+            if (CertificateType == ObjectTypeIds.EccBrainpoolP256r1ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccBrainpoolP384r1ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccCurve25519ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccCurve448ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccNistP256ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccNistP384ApplicationCertificateType)
+            {
+                return securityConfiguration.MinimumECCertificateKeySize;
+            }
+            else if (
+                CertificateType == ObjectTypeIds.RsaMinApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.RsaSha256ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.ApplicationCertificateType ||
```

Comment:
are ecc certs not subtypes of ApplicationCertificateType?

Reply:
Correct